### PR TITLE
SF-1932 - Store updated avatar URL in Auth0 user metadata

### PR DESCRIPTION
**Pull request by nigel-wells**
- Save any update to the Avatar URL (picture) in the user metadata in Auth0
- On receiving the user profile from Auth0, use the picture value from metadata if available
- Update presence information when the user doc changes
- Added return types
- Added a user doc to various tests that use the text component

- - -
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1772)

### CODE REVIEW

File: src/SIL.XForge/Services/UserService.cs [hunk 1]
```diff
@@ -69,9 +69,12 @@ await using (IConnection conn = await _realtimeService.ConnectAsync(curUserId))
             );
             await userDoc.SubmitJson0OpAsync(op =>
             {
+                string picture =
```
Comment (pmachapman) [inline src/SIL.XForge/Services/UserService.cs:72]: I found this block a little confusing on first glance (I had to double take to figure out what it was doing). One way you could make this more readable is to simplify this line as follows:
```csharp
string picture = (string)userProfile["user_metadata"]?["picture"];
```

**Review by pmachapman — CHANGES_REQUESTED** [review]
Looks good. Thank you for cleaning up the TypeScript warnings and removing deadwood as you were updating. Just one comment on a bit I found confusing.

**Review by nigel-wells — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1772)* status: 0 of 9 files reviewed, 1 unresolved discussion (waiting on @pmachapman)

___
*[`src/SIL.XForge/Services/UserService.cs` line 72 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1772#-NReTw0JdiGwA23GEj-D-r1-72:-NS3C3H2A1Ylf0Yy8YBw:b69u4km) ([raw file](https://github.com/sillsdev/web-xforge/blob/d64a6981f95dc4083ddb64a7d232c3bf1e9f2828/src/SIL.XForge/Services/UserService.cs#L72)):*
<details><summary><i>Previously, pmachapman (Peter Chapman) wrote…</i></summary><blockquote>

I found this block a little confusing on first glance (I had to double take to figure out what it was doing). One way you could make this more readable is to simplify this line as follows:
```csharp
string picture = (string)userProfile["user_metadata"]?["picture"];
```
</blockquote></details>

Ah neat, didn't realise we could do null checks like that. Does this same approach work for anything in C#? I assume it would work with more dimensions i.e. `(string)userProfile["user_metadata"]?["picture"]?["url"];` ?

**Review by pmachapman — APPROVED** [review]
> Reviewed 8 of 9 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/1772)* status: all files reviewed, 1 unresolved discussion (waiting on @nigel-wells)

___
*[`src/SIL.XForge/Services/UserService.cs` line 72 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/1772#-NReTw0JdiGwA23GEj-D-r1-72:-NS3CjJG3ISfisz42vgE:b-p4x4wl) ([raw file](https://github.com/sillsdev/web-xforge/blob/d64a6981f95dc4083ddb64a7d232c3bf1e9f2828/src/SIL.XForge/Services/UserService.cs#L72)):*
<details><summary><i>Previously, nigel-wells (Nigel Wells) wrote…</i></summary><blockquote>

Ah neat, didn't realise we could do null checks like that. Does this same approach work for anything in C#? I assume it would work with more dimensions i.e. `(string)userProfile["user_metadata"]?["picture"]?["url"];` ?
</blockquote></details>

It works for anything that is nullable in C# (strings, objects, etc...). And yes, it works multiple levels deep.

**Pull request merged**